services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedServiceConfig.java

```diff
@@ -139,6 +199,13 @@ public FileBasedDeviceBackend deviceBackend(final Vertx vertx) {
         final FileBasedRegistrationService registrationService = new FileBasedRegistrationService(vertx);
         registrationService.setConfig(registrationProperties());
 
+        final AutoProvisioner autoProvisioner = new AutoProvisioner(registrationService, registrationService);
+        autoProvisioner.setVertx(vertx);
+        autoProvisioner.setDownstreamSenderFactory(downstreamSenderFactory(vertx));
+        autoProvisioner.setConfig(autoProvisionerConfigProperties());
```

Comment:
It looks to me as if invoking `autoProvisioner.setTracer()` and `autoProvisioner.setTenantInformationService()` is missing here.

Reply:
Have you tried defining `AutoProvisioner` as a Bean here instead, to make use of the Autowired annotations in the `AutoProvisioner` class?

Reply:
Yes, to make it consistent I set them explicitly here as well.

Reply:
It turned out that the _Tracer_ instance is defined in the parent application context. The _TenantInformationService_ was missing though.

Reply:
Good catch, thanks! 😄 

It turned out that the _Tracer_ bean has to be taken from the root application context of the file based device registry and the TenantInformationService wasn't defined in the context so far leading to the NOOP service being used.

I could define AutoProvisioner as a bean instead, but then it must be refactored not to take any constructors arguments and use setters instead. 

But as it seems @sophokles73 changed the declaration explicitly to use setters in 17179adb6933c0eb3d1eeeccc26518e27297d977 I am wondering if we should do this in the scope of this PR or re-discuss this in a separate issue and change the "bean-declaration-style" not just for the beans affected in this PR but for all beans (and for all other registries as well)...

Reply:
Ok, yes, we can discuss a possible refactoring here once all registry changes have been merged.